filecoin-proofs/tests/api.rs

```diff
@@ -79,6 +80,156 @@ fn seal_lifecycle<Tree: 'static + MerkleTreeTrait>(sector_size: u64) -> Result<(
+            let cur = entry.expect("reading directory failed");
+            let entry_path = cur.path();
+            let entry_str = entry_path.to_str().expect("failed to get string from path");
+            if entry_str.contains("data-layer") {
+                Some(entry_path.clone())
+            } else {
+                None
+            }
+        })
+        .collect()
+}
+
+fn clear_cache_dir_keep_data_layer(cache_dir: &tempfile::TempDir) {
+    for entry in fs::read_dir(&cache_dir).expect("faailed to read directory") {
+        let entry_path = entry.expect("failed get directory entry").path();
+        if entry_path.is_file() {
+            // delete everything except the data-layers
+            if !entry_path
+                .to_str()
+                .expect("failed to get string from path")
+                .contains("data-layer")
+            {
+                fs::remove_file(entry_path).expect("failed to remove file")
+            }
+        }
+    }
+}
+
+#[test]
+fn test_resumable_seal_skip_proofs() {
+    run_resumable_seal(true, 0);
+    run_resumable_seal(true, 1);
+}
+
+#[test]
+#[ignore]
+fn test_resumable_seal() {
+    run_resumable_seal(false, 0);
+    run_resumable_seal(false, 1);
+}
+
+/// Create a seal, delete a layer and resume
+///
+/// The current code works on two layers only. The `layer_to_delete` specifies (zero-based) which
+/// layer should be deleted.
+fn run_resumable_seal(skip_proofs: bool, layer_to_delete: usize) {
+    init_logger();
+
+    let sector_size = SECTOR_SIZE_2_KIB;
+    let rng = &mut XorShiftRng::from_seed(TEST_SEED);
+    let prover_fr: DefaultTreeDomain = Fr::random(rng).into();
+    let mut prover_id = [0u8; 32];
+    prover_id.copy_from_slice(AsRef::<[u8]>::as_ref(&prover_fr));
+
+    let (mut piece_file, piece_bytes) =
+        generate_piece_file(sector_size).expect("failed to generate piece file");
+    let sealed_sector_file = NamedTempFile::new().expect("failed to created sealed sector file");
+    let cache_dir = tempfile::tempdir().expect("failed to create temp dir");
+
+    let arbitrary_porep_id = rng.gen();
+    let config = porep_config(sector_size, arbitrary_porep_id);
+    let ticket = rng.gen();
+    let sector_id = rng.gen::<u64>().into();
+
+    // First create seals as expected
+    run_seal_pre_commit_phase1::<Tree>(
+        config,
+        prover_id,
+        sector_id,
+        ticket,
+        &cache_dir,
+        &mut piece_file,
+        &sealed_sector_file,
+    )
+    .expect("failed to run seal pre commit phase1");
+    let layers = get_layer_file_paths(&cache_dir);
+    assert_eq!(layers.len(), 2, "not all expected layers were created");
+
+    // Delete one layer, keep the other
+    clear_cache_dir_keep_data_layer(&cache_dir);
+    std::fs::remove_file(&layers[layer_to_delete]).expect("failed to remove layer");
+    let layers_removed = get_layer_file_paths(&cache_dir);
+    assert_eq!(layers_removed.len(), 1, "found more than one layer");
```

Comment:
This name is somewhat confusing. As written, I assume `layers_removed` would contain the layers which were removed, not the layers *not* removed (i.e. those remaining).

The assertion doesn't clarify it for me, either, since with only two layers hardcoded as the test size, either the removed or remaining layers should have length 1.

Also, I'm skeptical of the assertion, since it could also fail if no layers were found, for whatever reason.

Reply:
> Also, I'm skeptical of the assertion, since it could also fail if no layers were found, for whatever reason.

I think it should fail if no layers are found as we are testing the resuming with existing layers. I'll reword the assertion message though.